research/conditional/utils/layer_manager.py

```diff
@@ -87,7 +87,11 @@ def log(self, step):
 
         for verbosity_level in verbosity_levels:
             for block_name, layer in self._layers:
-                if isinstance(layer, LoggingLayer):
+                if isinstance(layer, LoggingLayer) or (
+                    isinstance(layer, torch.distributed.fsdp.FullyShardedDataParallel)
+                    and isinstance(layer._fsdp_wrapped_module, LoggingLayer)
```

Comment:
so does the fsdp wrappaer inherit the `log` method? otherwise, it doesnt work. Or am i missing sth

Reply:
![image](https://github.com/llm-random/llm-random/assets/70006947/3c057566-2496-46a6-ae54-92e422dbb6d8)